cpp/include/cudf/aggregation.hpp

```diff
@@ -599,6 +609,245 @@ std::unique_ptr<Base> make_udf_aggregation(udf_type type,
                                            std::string const& user_defined_aggregator,
                                            data_type output_type);
 
+/**
+ * @brief The interface for host-based UDF implementation.
+ *
+ * An implementation of host-based UDF needs to be derived from this base class, defining
+ * its own version of the required functions.
+ */
```

Comment:
It might be helpful to include in the class documentation a list of functions or optional customization points (e.g. `get_required_data`) that users need to override or an example of a minimal use case (e.g., groupby max) for clarity.